presto-main/src/main/java/com/facebook/presto/operator/PageSourceOperator.java

```diff
@@ -71,6 +73,12 @@ public boolean isFinished()
     }
 
     @Override
+    public ListenableFuture<?> isBlocked()
+    {
+        return toListenableFuture(pageSource.isBlocked());
```

Comment:
That's likely a bigger project and is a question for @dain. It would cleanup places like this, but `ListenableFuture` is arguably a better API, so I don't know if he wants to change it.